### PR TITLE
added publish quiz command and simple quiz iframe

### PR DESCRIPTION
**Pull request by tuanh118**
- Added `/publish_quiz` admin command to push an LMS quiz to all users who are enrolled in the bootcamp course
- Added Quiz Iframe that listens to `publish_quiz` events

### CODE REVIEW

File: backend/src/auth/authentication_server.ts [hunk 1]
```diff
@@ -6,10 +6,15 @@ type LMS_DECODED_TOKEN = {
...
 
-export default class AuthenticationServer {
-  async login(username: string, password: string): Promise<LMS_DECODED_TOKEN> {
-    const apiBaseUrl = process.env.LMS_BASE_URL;
+type AUTH_RESPONSE = {
+  preferred_username?: string;
+  is_enrolled: boolean;
+}
 
+const COURSE_ID = "course-v1%3ASTEAMforVietnam%2BScratch101%2B2020_08";
```
Comment (vudung45) [inline backend/src/auth/authentication_server.ts:14]: Dùng  `encodeURIComponent` đc không anh?

File: backend/src/auth/authentication_server.ts [hunk 2]
```diff
@@ -22,16 +27,43 @@ export default class AuthenticationServer {
...
+          const user = userInfo?.preferred_username;
           if (!user) {
             return reject("Invalid JWT token received");
           }
 
-          resolve(jwt.decode(response.data.access_token) as LMS_DECODED_TOKEN);
+          let isEnrolled = false;
+          this.isUserEnrolled(user, response.data.access_token)
+            .then(enrollmentResponse => isEnrolled = enrollmentResponse)
+            .catch(_ => isEnrolled = false);
```
Comment (vudung45) [inline backend/src/auth/authentication_server.ts]:
```suggestion
          await this.isUserEnrolled(user, response.data.access_token)
            .then(enrollmentResponse => isEnrolled = enrollmentResponse)
            .catch(_ => isEnrolled = false);
```

Comment (tuanh118) [reply backend/src/auth/authentication_server.ts]: a vua fixed doan nay roi

File: backend/src/auth/authentication_server.ts (hunk 1 above)
Comment (tuanh118) [reply backend/src/auth/authentication_server.ts]: unencoded works as well. Pushed the change

**Review by vudung45 — APPROVED** [review]

File: frontend/src/Components/QuizModal.tsx [hunk 3]
```diff
@@ -16,22 +19,47 @@ export default function QuizModal({
...
+      label: userData.username,
+    });
     window.open("https://courses.steamforvietnam.org/login", "_blank");
   };
 
   const onCloseClick = () => {
     // TODO: get iframe content to check FINISH status
-    // TODO: emit GA event
-    onModalClose && onModalClose();
+    const shouldClose = window.confirm(`Hãy chắc chắn bạn đã trả lời câu hỏi trước khi ẩn nó đi. Câu hỏi đã được ẩn sẽ không thể trả lời lại.
```
Comment (tuanh118) [inline frontend/src/Components/QuizModal.tsx:45]: thay "ẩn" bằng "đóng" để rõ nghĩa hơn :D

Comment (manhhung741) [reply frontend/src/Components/QuizModal.tsx:45]: +1

Comment (namdaoduy) [reply frontend/src/Components/QuizModal.tsx]: Fixed

**Pull request merged**